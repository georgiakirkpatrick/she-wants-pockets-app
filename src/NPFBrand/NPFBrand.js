import React from 'react'
import FormButton from '../FormButton/FormButton'
import FormDropdown from '../FormDropdown/FormDropdown'
import FormPage from '../FormPage/FormPage'
import FormPopUp from '../FormPopUp/FormPopUp'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import FormTextInput from '../FormTextInput/FormTextInput'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import NPFFooter from '../NPFFooter/NPFFooter'
import sizes from '../SIZES'

const NPFBrand = props => {
    const newBrandPopUp = () => {
        if (props.brandPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const makeBrandOptions = () => {
        const defaultOption = {
            id: 'blank-brand',
            text: 'Select a brand'
        }
        
        const brands = props.brands.map(brand => ({
            id: brand.id,
            text: brand.english_name
        }))

        return [
            defaultOption,
            ...brands
        ]
    }

    const changeBrandFields = event => {
        const values = {...props.newBrandFields}
        values[event.target.name] = event.target.value
        props.setNewBrandFields(values)
    }

    const handleBrandPopClose = () => {
        props.setBrandPopUp(false)
    }

    const createBrand = () => {}

    const handleBrandPopSubmit = () => {
        createBrand()

        handleBrandPopClose()
    }

    return (
        <div id='NPFBrand'>
            <FormPage title='Brand'>
                <FormDropdown 
                    currentBrandId={props.currentBrandId}
                    handleChange={(event) => {
                        props.setCurrentBrandId(event.currentTarget.id)
                    }}
                    id='brand'
                    name='brand'
                    prompt='Which brand made this product?'
                    options={makeBrandOptions()}
                />
                <FormPromptWithSub
                    prompt='Brand not listed above?'
                    promptSubtitle=''
                />
                <FormButton
                    buttonText='ADD A BRAND' 
                    handleClick={() => props.setBrandPopUp(true)} 
                    type='button'
                />
                <FormPopUp 
                    id='add-brand'
                    status={newBrandPopUp()}
                    title='New Brand'
                    close={() => handleBrandPopClose()}
                    submit={() => handleBrandPopSubmit()}
                    buttonText='CREATE BRAND'
                >
                    <FormPromptWithSub
                        prompt='Enter the following details to add a brand.'
                    />
                    <FormTextInput
                        id='brandName'
                        name='brandName'
                        prompt='Brand name'
                        currentValue={props.newBrandFields['brandName']} 
                        handleChange={event => changeBrandFields(event)}
                    />
                    <FormUrlInput 
                        id='brandWebsite'
                        name='brandWebsite'
                        prompt='Brand website'
                        currentValue={props.newBrandFields['brandWebsite']} 
                        handleChange={event => changeBrandFields(event)}
                    />
                    <FormDropdown
                        id='brandCurrency'
                        name='brandCurrency'
                        prompt='Which currency does the brand list prices in?'
                        currentValue={props.newBrandFields['randCurrency']} 
                        handleChange={event => changeBrandFields(event)}
                        options={props.currencies}
                    />
                    <FormDropdown
                        id='brandSizeSystem'
                        name='brandSizeSystem'
                        prompt='Which sizing system does the brand use?'
                        currentValue={props.newBrandFields['brandSizeSystem']} 
                        handleChange={event => changeBrandFields(event)}
                        options={sizes.systems}
                    />
                </FormPopUp>
            </FormPage>
            <NPFFooter buttons='prevNext' previousButton={() => props.setPage(props.currentPage - 1)} nextButton={() => props.setPage(props.currentPage + 1)} />
        </div>
    )
}

NPFBrand.defaultProps = {
    currentBrandId: 0,
    setCurrentBrandId: () => {},
    newBrandFields: {
        brandName: '',
        brandWebsite: '',
        brandCurrency: '',
        brandSizeSystem: ''
    },
    setNewBrandFields: () => {},
    brands: [],
    setBrands: () => {},
    brandPopUp: false,
    setBrandPopUp: () => {},
    currentPage: 0,
    setBrandId: () => {},
    setPage: () => {},
    currencies: []
}

export default NPFBrand