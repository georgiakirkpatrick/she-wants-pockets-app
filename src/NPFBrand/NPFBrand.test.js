import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NPFBrand from './NPFBrand'

describe('NPFBrand', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFBrand />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<NPFBrand />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})