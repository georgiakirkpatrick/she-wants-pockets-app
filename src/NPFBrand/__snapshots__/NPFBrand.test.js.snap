// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFBrand render the UI as expected 1`] = `
<div
  id="NPFBrand"
>
  <section
    className="FormPage"
  >
    <h1>
      Brand
    </h1>
    <form>
      <div
        className="FormDropdown"
      >
        <label
          htmlFor="brand"
        >
          Which brand made this product?
        </label>
        <select
          name="brand"
          onChange={[Function]}
        >
          <option
            id="blank-brand"
          >
            Select a brand
          </option>
        </select>
      </div>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Brand not listed above?
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <button
        className="FormButton"
        onClick={[Function]}
        type="button"
      >
        ADD A BRAND
      </button>
      <section
        className="FormPopUp__pop-up"
        id="add-brand"
      >
        <button
          className="FormPopUp__close-button"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 fa-lg "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
        <h1>
          New Brand
        </h1>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Enter the following details to add a brand.
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <div
          className="FormText"
        >
          <label
            htmlFor="brandName"
          >
            Brand name
          </label>
          <input
            defaultValue=""
            id="brandName"
            name="brandName"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="FormUrlInput"
        >
          <label
            htmlFor="brandWebsite"
          >
            Brand website
          </label>
          <input
            defaultValue=""
            id="brandWebsite"
            name="brandWebsite"
            onChange={[Function]}
            type="url"
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="brandCurrency"
          >
            Which currency does the brand list prices in?
          </label>
          <select
            name="brandCurrency"
            onChange={[Function]}
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="brandSizeSystem"
          >
            Which sizing system does the brand use?
          </label>
          <select
            name="brandSizeSystem"
            onChange={[Function]}
            value=""
          >
            <option
              id="us-sizes"
            >
              US
            </option>
            <option
              id="france-sizes"
            >
              France
            </option>
            <option
              id="italy-sizes"
            >
              Italy
            </option>
            <option
              id="uk-sizes"
            >
              UK
            </option>
            <option
              id="japan-sizes"
            >
              Japan
            </option>
            <option
              id="other-sizes"
            >
              Other
            </option>
          </select>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          CREATE BRAND
        </button>
      </section>
    </form>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter_previous"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 NPFFooter_arrow"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter_next"
      onClick={[Function]}
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 NPFFooter_arrow"
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
